upstream/upstream.go

```diff
@@ -693,7 +693,11 @@ func (u *Upstream) detectFeatures() error {
 			u.supportedNetworkIds[util.EvmNetworkId(cfg.Evm.ChainId)] = true
 			u.supportedNetworkIdsMu.Unlock()
 		}
-		// TODO evm: check full vs archive node
+
+		if cfg.Evm.MaxAvailableRecentBlocks == 0 && cfg.Evm.NodeType == common.EvmNodeTypeFull {
+			cfg.Evm.MaxAvailableRecentBlocks = 128
```

Comment:
Is this number universally correct? on all major eth clients etc

If this number is too high is better than too low, as we'll just failover if node is indeed missing the data

Reply:
`geth: By default, state for the last 128 blocks kept in memory`
https://github.com/ethereum/go-ethereum/releases/tag/v1.8.0

`Also known as pruned node, a full node maintains the blockchain's current state and data on the network, excluding trace data for transactions that are beyond the most recent 128 blocks`
https://erigon.gitbook.io/erigon/basic-usage/usage/type-of-node#full-node-or-pruned-node

Reply:
there was an interesting comment in `Erigon` docs e.g. related to full-nodes. could it be that the full will be marked as full, but can store beyond 128 blocks using `prune` command?

Reply:
I see, ok so 128 is safest default then